src/renderer/components/data-settings/data-settings.js

```diff
@@ -125,9 +125,43 @@ export default Vue.extend({
           let textDecode = new TextDecoder('utf-8').decode(data)
           textDecode = textDecode.split('\n')
           textDecode.pop()
+          textDecode = textDecode.map(data => JSON.parse(data))
+
+          // TODO: Move this function to a better place
```

Comment:
Is this TODO for future improvements, or something that you originally intended to handle before making the PR?

Reply:
I'm just not familiar enough with the architecture to understand where a function like that would be best placed. So no I had no intention of moving the function, but if in order to get this PR accepted, it needs to be moved, I'm willing to do it. Someone just needs to tell me where exactly it needs to go.

Reply:
You are correct.  This function should be moved.  It can simply be moved towards the bottom of the methods in between `exportHistory` and `getChannelInfoInvidious`.  Within the `importFreeTubeSubscriptions` function, you would call it with the syntax `this.convertOldFreeTubeFormatToNew()` instead of the way it's called now.  You may need to include an `await` method as well to prevent the logic from being skipped due to it being asynchronous.

If you come across any issues with moving it or unsure of the syntax, please let me know.